packages/schema/test/Schema/brand.test.ts

```diff
@@ -117,7 +117,10 @@ describe("brand", () => {
    └─ Expected IntegerFromString (an integer), actual 1.1`,
           ParseResult.parseError(
             new ParseResult.Refinement(
-              IntegerFromString.ast as any,
+              // FIXME: the `from` is actually a Number AST instead of a Transformation
+              // expected IntegerFromString.ast as any
+              // console.log(e[0].meta.error.ast)
+              Object.assign(IntegerFromString.ast, { from: S.Number.ast }) as any,
```

Comment:
@gcanti could please check locally to make sure I'm not hallucinating? The `e[0].meta.error.ast` really seems to be a refinement from a number instead of a refinement from the `S.NumberFromString` transformation.

Reply:
the test is wrong, IMO should be
```ts
  it("the constructor should throw on invalid values", () => {
    const IntegerFromString = S.NumberFromString.pipe(
      S.int({ identifier: "IntegerFromString" }),
      S.brand("Int")
    )
    expect(IntegerFromString(1)).toEqual(1)
    try {
      IntegerFromString(1.1)
      assert.fail("expected `IntegerFromString(1.1)` to throw an error")
    } catch (e: any) {
      const ast = S.typeSchema(IntegerFromString).ast as AST.Refinement
      expect(e).toStrictEqual(
        Brand.error(
          `IntegerFromString
└─ Predicate refinement failure
   └─ Expected IntegerFromString (an integer), actual 1.1`,
          ParseResult.parseError(new ParseResult.Refinement(ast, 1.1, "Predicate", new ParseResult.Type(ast, 1.1)))
        )
      )
    }
  })
```

Reply:
perfect, that passes, thanks 👍 (updated)